src/entities/permit2Helper/index.ts

```diff
@@ -195,6 +196,46 @@ export class Permit2Helper {
         return signPermit2(input.client, input.owner, spender, details);
     }
 
+    static async signInitBufferApproval(
+        input: InitBufferBuildCallInput & {
+            client: PublicWalletClient;
+            owner: Address;
+            nonces?: number[];
+            expirations?: number[];
+        },
+    ): Promise<Permit2> {
+        if (input.nonces && input.nonces.length !== 2) {
+            throw new Error("Nonces length doesn't match amountsIn length");
+        }
+        if (input.expirations && input.expirations.length !== 2) {
+            throw new Error(
+                "Expirations length doesn't match amountsIn length",
+            );
+        }
+        const spender = BALANCER_BUFFER_ROUTER[input.chainId];
+        const details: PermitDetails[] = [
+            await getDetails(
+                input.client,
+                input.wrappedAmountIn.token.address,
+                input.owner,
+                spender,
+                input.wrappedAmountIn.amount,
+                input.expirations ? input.expirations[0] : undefined,
+                input.nonces ? input.nonces[0] : undefined,
+            ),
+            await getDetails(
+                input.client,
+                input.underlyingAmountIn.token.address,
+                input.owner,
+                spender,
+                input.underlyingAmountIn.amount,
+                input.expirations ? input.expirations[1] : undefined,
+                input.nonces ? input.nonces[1] : undefined,
+            ),
+        ];
```

Comment:
i would have tried to map this, but i like your approach better given the circumstances

Reply:
Yeah, I though of having an array of amountsIn and stating somewhere that index 0 was the wrapped token and index 1 was underlying token, but that felt a bit error prone.
Since it's always 2 tokens only, having them explicitly declared like this felt like a better approach.